skimage/measure/_find_contours_cy.pyx

```diff
@@ -93,6 +100,14 @@ def iterate_and_store(double[:, :] array,
             coords[0] += 1
             coords[1] = 0
 
+        # Skip this square if any of the four input values are masked out.
+        if not (mask[r0, c0] and mask[r0, c1] and
+                mask[r1, c0] and mask[r1, c1]):
+            continue
+
+        # Skip this square if any of the four input values are NaN.
+        if npy_isnan(ul) or npy_isnan(ur) or npy_isnan(ll) or npy_isnan(lr):
```

Comment:
Instead of checking for nans here, we could make a copy of `mask` and do `mask &= ~np.isnan(array)` before we start, as suggested by @jni. I imagine this would be slightly faster (since we'd read each array value for nan checks once instead of four times) but use slightly more memory to store the copy. I have no idea whether this is likely to matter much - I'm new around here, so happy to go with whatever you all think is best.

Reply:
The best way to figure out which is faster is to benchmark

https://scikit-image.org/docs/dev/contribute.html#benchmarks

Reply:
I think you'll also find that what is faster for the small arrays, will not be faster for large arrays.

Reply:
To get a quick ballpark estimate you can also use [ipython's %timeit command](https://ipython.org/ipython-doc/3/interactive/magics.html#magic-timeit) (in case you didn't know already).